cli/internal/codegen/testdata/expected_golang.go

```diff
@@ -402,6 +402,9 @@ func (c *svcClient) Webhook(ctx context.Context, a string, b string, request *ht
 		return nil, fmt.Errorf("unable to parse api url: %w", err)
 	}
 	request = request.WithContext(ctx)
+	if request.Method == "" {
+		request.Method = "POST"
+	}
```

Comment:
We override the URL object passed to a raw endpoint already as that's set at the API level. So we don't expect request objects passed in to have the URL set (and if they are, we ignore them).

The same applies for the method, the API helper function I'm using in the client knows how to call the raw endpoint, so I expect it to be able to call said endpoint with what ever request object provide.

Reply:
i.e. we should be able to make a request like this:

https://github.com/encoredev/encore/pull/229/files#diff-cd3cd9de3bee67c24f7159ff7ece28f7ce5b0b088c2dce350d337dd9d49e4d4bR178-R185

Reply:
Well that’s the case because we’re providing the convenience of accepting path parameters in the signature.

But method seems different: in valid Go code it’s not really possible to create an `*http.Request` without a valid method, is it? There are lots of field we expect you to have set on the Request you pass us, why does method deserve special treatment?

I guess my question is: what was the scenario that motivated this? And was that code valid or invalid?

Reply:
Should we accept just a raw object constructed like that, though? The standard constructor for outgoing requests does a whole lot more: https://cs.opensource.google/go/go/+/refs/tags/go1.18.2:src/net/http/request.go;l=856

Reply:
> I guess my question is: what was the scenario that motivated this? And was that code valid or invalid?

In this example:

```go
		req := &http.Request{
			Header: http.Header{
				"X-Test-Header": []string{"test"},
			},
			Body: io.NopCloser(strings.NewReader("this is a test body")),
		}

		rsp, err := api.Test.RawEndpoint(ctx, "hello", req)
```

A _valid_ request object is constructed, with a header and body provided - with is what the Raw endpoint needs in this case. Our generated client will add anything else needed to call that endpoint, including;
- Auth token to the header
- The URL to the handler
- The method required by the handler

Asking the user to include the method or URL requests in the question why are they even using the generated client in that case?

It's also worth noting the documentation for `http.Header` states method can be empty - and will be assumed to be `GET` in that case. In our case; we're actually assuming it to be the method required by that API.

Reply:
> Should we accept just a raw object constructed like that, though? The standard constructor for outgoing requests does a whole lot more

I'm conflicted here; I'd argue yes - because we can populate the rest.

We could create a `api.Test.NewRawEndpointRequest(body io.Reader) *http.Request` method, which we then expect you to use with the `api.Test.RawEndpoint(ctx context.Context, id string, req *http.Request)` method.

But then;
1. How could we easily support path params without the using having to do the encoding themselves?
2. Should things like the auth token be added to that returned request object
3. If we return that, we still want the request to be used through the API client, so other options in the baseClient can be applied - having a raw req object though would be tempting to simply execute it using any round tripper avaliable.

Reply:
Okay, since `(*http.Request).Method`'s documentation specifies it can be empty that's a compelling reason to support a default value. But I don't think we should be making "smart decisions" about sensible defaults for raw requests, because the whole point of raw requests is to enable full control over the request. So we should be clear about which fields we override or support empty values for. I don't think we should document that you can pass in a raw `&http.Request{Header: foo, Body: bar}` literal, because that's in general not a supported way of using `(http.RoundTripper).Do` as far as I know.